### ISSUE TITLE
Android example not working

### ISSUE THREAD

**Issue by lesniakania** [magus/react-native-facebook-login#72]
For sure `index.android.js` is missing, but it's not all unfortunatelly.
```
react-native-facebook-login/example :) [master] $ react-native run-android
Command `run-android` unrecognized. Did you mean to run this inside a react-native project?
```

**Comment by mbrochh** [comment]
are you sure? android example is working for me even better than the iOS one...

**Comment by lesniakania** [comment]
So how can I run it? As I mentioned running `react-native run-android` from `example` directory results with `Command `run-android` unrecognized. Did you mean to run this inside a react-native project?`

Do I need to have specific `react-native` version globally for that?

**Comment by mbrochh** [comment]
Hmm I'm not sure about this. Never tried to run the example project of this repo. Did you run `npm install` before trying `run-android`?

**Comment by lesniakania** [comment]
Yes, I run `npm install`. I mean, if you have example project in the repo and you have android setup there (and your library support Android) I would expect that example would work for Android too ;]

**Comment by lesniakania** [comment]
I mean, I'm asking because I have problems with Android setup (while iOS works like charm).
I don't have any errors, I actually have no clue what's going on.
That's why I was interested how it works in your example.

**Comment by magus** [comment]
Anna, unfortunately the example is for iOS works but the Android version may indeed be out of date. I'll leave this open for the time being as a record that it needs to be updated. 

Most likely the necessary changes are similar to what was updated in the README as this was a recent issue when people used react-native 0.18+ with this repo.

https://github.com/magus/react-native-facebook-login/commit/b7884fd50bf68f7a84b1b81f3eb49af0b7c5f262

https://github.com/magus/react-native-facebook-login/pull/60/files

**Comment by magus** [comment]
@lwhiteley @satya164

**Comment by lwhiteley** [comment]
i dont think i included or updated the example project actually

Plus there is no **index.android.js** in the example directory, so i dont think it would work

**Comment by lesniakania** [comment]
Yeah, that's what I'm talking about : ] But you have `android` directory in the example and you support Android, so it's confusing. Just saying it would be nice to have, I actually managed to get it working.

**Comment by magus** [comment]
@lesniakania want to share your example? I am sure some people would appreciate it!

**Comment by lesniakania** [comment]
I was thinking about this, I will try to find some time : ]

**Comment by saeed2402** [comment]
Has anybody been able to work this out? I have been struggling to implement facebook login in Android for a week now, any help is greatly appreciated..

**Comment by satya164** [comment]
@saeed2402 Have you checked the official SDK? https://github.com/facebook/react-native-fbsdk

**Comment by saeed2402** [comment]
@satya164  Yes, I did. I guess (!) I followed all the steps, but what should go in my index.android.js file??

**Comment by lwhiteley** [comment]
@saeed2402 not sure what your issue is but let me know if you followed the instructions https://github.com/magus/react-native-facebook-login/tree/master/android

once you have all the requirements such as creating a facebook app on developer.facebook.com and follow the setup instructions then you shouldn't have any issues.

same would go for the official fbsdk repo

**Comment by saeed2402** [comment]
Thanks @lwhiteley . My biggest problem right now is that I don't know what should be in my index.android.js file. I followed all the steps in your repo, but can't get it work even though I tried different different things in index.android.js file.
Do you have a sample for that file that goes with your project?

**Comment by lwhiteley** [comment]
https://github.com/magus/react-native-facebook-login/tree/master/android#step-6---include-in-javascript

you just need to import the module and put the component where you need it in your component's render function

**Comment by lwhiteley** [comment]
```js
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View
} from 'react-native';

import {FBLogin, FBLoginManager} from 'react-native-facebook-login';

class myApp extends Component {
  render() {
    return (
      <View style={styles.container}>
        <FBLogin
             ref={(fbLogin) => { this.fbLogin = fbLogin }}
            loginBehavior={FBLoginManager.LoginBehaviors.Native}
            permissions={["email","user_friends"]}
            onLogin={function(e){console.log(e)}}
            onLoginFound={function(e){console.log(e)}}
            onLoginNotFound={function(e){console.log(e)}}
            onLogout={function(e){console.log(e)}}
            onCancel={function(e){console.log(e)}}
            onPermissionsMissing={function(e){console.log(e)}}
          />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  }
});

AppRegistry.registerComponent('myApp', () => myApp);
```

**Comment by saeed2402** [comment]
Thanks again. I started with a fresh project and followed the instructions. Better result this time: I can at lease see "login with facebook" button.
But my app crashes the moment I clock on it. After running debugger, I realized it's due to facebook not having been set up on my emulator. Any idea how to do that using facebook android sdk 4.16 on mac? All posts I could find so far reference older sdks, specially talking about FBAndroid.apk that I can't find..

**Comment by lwhiteley** [comment]
@saeed2402 

please create a issue for this and ill help you debug.
ill have to test this but naturally it should fallback to a web browser unless the sdk had some changes i didnt test yet.

can you set the behaviour to web and see what happens for now

```js
loginBehavior={FBLoginManager.LoginBehaviors.Web}
```

**Comment by saeed2402** [comment]
Tried to change the behavior to web, same result. For your reference, below is the output in the debugger:

Console was cleared
index.android.bundle:41249Running application "androidS" with appParams: {"initialProps":{},"rootTag":1}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
index.android.bundle:81456Triggering 'onLoginNotFound' event
index.android.bundle:1507Object {declinedPermissions: Array[0], provider: "facebook", message: "No user found", type: "cancel"}
